pymongo/client_session.py

```diff
@@ -893,10 +913,10 @@ def advance_operation_time(self, operation_time: Timestamp) -> None:
             raise TypeError("operation_time must be an instance of bson.timestamp.Timestamp")
         self._advance_operation_time(operation_time)
 
-    def _process_response(self, reply):
+    def _process_response(self, reply: Mapping[str, Any]) -> None:
         """Process a response to a command that was run with this session."""
-        self._advance_cluster_time(reply.get("$clusterTime"))
-        self._advance_operation_time(reply.get("operationTime"))
+        self._advance_cluster_time(cast(Mapping[str, Any], reply.get("$clusterTime")))
+        self._advance_operation_time(cast(Timestamp, reply.get("operationTime")))
```

Comment:
These casts should not be needed after the changes above.

Reply:
Done.